ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1375,6 +1471,16 @@ protected void maybeProcessCallbacks() {
     @Override
     void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInfo messageInfo)
         throws IOException, MessageFormatException {
+
+      if (isMetadataCacheHit()) {
+        getOptions().ageAtAccessTracker.trackAgeAtAccess(serverBlobProperties.getCreationTimeInMs());
+        handleMetadataBlob(null, blobInfo.getUserMetadata(), null);
+        // Do not call updateTtlIfRequired(). If the blob expires, the frontend will find out from backend during the next GET request
+        // TODO: Deserialize payload and do basic validations to check if cached blob properties are correct.
+        // TODO: Should we decrypt ? Probably no.
+        return;
+      }
+
```

Comment:
handleBody is the method to call when handling response from server node, it shouldn't have anything to do with cache.

Reply:
this is the handleBody for the first chunk - metadata chunk. So if we have a cacheHit, we don't need to do anything below this clause. Datachunks have a different handleBody